lesson3.md

```diff
@@ -0,0 +1,837 @@
+  let greeting = "Hello my dear";
+}
+
+console.log(greeting); // ERROR!!!!
+```
+
+
+---
+
+
+### Code Blocks: {}
+
+Correct solution:
+
+```js
+let name = "John Doe";
+
+let greeting;
+if (name === "John Doe") {
+  greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  greeting = "Hello my dear";
+}
+
+console.log(greeting); // "Hi dude"
+```
+
+
+---
+
+
+### Quiz
+
+What does it print?
+
+```js
+let food = "broccoli";
+
+if (food === "pizza") {
+    console.log("yum yum");
+}
+```
+
+
+---
+
+
+### Quiz
+
+What does it print?
+
+```js
+let age = 42;
+
+if (age >= 18) {
+    console.log("you are allowed to drive a car");
+}
+```
+
+
+---
+
+
+### More on the quiz
+
+If we can get in the cinema, log to the console: Nice! we can get in.
+
+
+If we can't get in the cinema, log to the console: We can't get in :(
+<!-- .element: class="fragment" -->
+
+
+This pattern of doing something when a variable is `true` and something else when the variable is `false` if very common, that is why we have `if else`.
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### If...Else
+
+
+
+```js
+if (condition) {
+  // some code
+  // will execute if condition is true
+} else {
+  // other code
+  // will execute if condition is false
+}
+```
+
+
+---
+
+
+### If...Else example
+
+```js
+if (temperature > 25) {
+  console.log("Go swimming");
+} else {
+  console.log("Go biking");
+}
+```
+
+
+---
+
+
+### More on the quiz
+
+Lets go back to our code and use `if else`.
+
+
+---
+
+
+### If...Else-If...Else example
+
+```js
+if (day === 'Saturday') {
+  console.log('Do all the shopping.');
+} else if (day === 'Sunday') {
+  console.log('Relax!');
+} else {
+  console.log('Wake up and go to work!');
+}
+```
+
+
+---
+
+
+### Exercise
+
+<!-- .slide: style="font-size:80%" -->
+
+* Make three if statements
+  * Make a "Good day" greeting if the hour is less than 18
+  * Make a statement that only says “Hello” to someone with your name.
+  * Make a statement that prints "I am younger" if your age is less than your neighbor's
+* Describe the weather based on the temperature:
+  * Above 30
+  * Above 20
+  * Above 5
+  * Everything else
+
+
+---
+
+
+### Is this correct?
+
+```js
+let temperature = 7;
+
+if (temperature > 20) {
+  console.log("it's warm");
+} else if (temperature > 30) {
+  console.log("too hot!!!");
+} else if (temperature > 5) {
+  console.log("perfect.");
+} else {
+  console.log("it's a bit chilly.")
+}
+```
+
+Try with a temperature of 35, what does it print? <!-- .element: class="fragment" -->
+
+
+---
+
+
+### Correct solution
+
+```js
+let temperature = 35;
+
+if (temperature > 30) {
+  console.log("too hot!!!");
+} else if (temperature > 20) {
+  console.log("it's warm");
+} else if (temperature > 5) {
+  console.log("perfect.");
+} else {
+  console.log("it's a bit chilly.")
+}
+```
+
+
+---
+
+
+### Let's code!
+
+Harald invented a game where the player with the highest value of his height (in cm) plus five times his age wins
```

Comment:
```suggestion
Abdullah invented a game where the player with the highest value of his height (in cm) plus five times his age wins
```

Reply:
YOU MADE IT PERSONAL BY PUTTING MY NAME THERE LAST SEMESTER.

I WILL TAKE IT WITH ME TO THE GRAVE.

Reply:
Well, you _did_ invent that game, after all. Games involving body height are not high on my list ;)